### PR TITLE
Networking changes to better support dhcp,mtu for virtual networks an…

### PR DESCRIPTION
**Pull request by muresan**
…d host network interfaces

### Summary and Scope

- Fixes:
- Requires:
- Relates to: Improve networking options for virtual networks and host network interfaces

#### Issue Type

- Bugfix Pull Request
- Docs Pull Request
- RFE Pull Request

### Prerequisites

- [ ] I have included documentation in my PR (or it is not required)
- [x] I tested this on internal system (if yes, please include results or a description of the test)
Applied dev successfully, tested mtu, dhcp for both virtual and macvtap.
- [ ] I tested this on a vshasta system (if yes, please include results or a description of the test)

### CODE REVIEW

File: dev/inventory.yaml [hunk 1]
```diff
@@ -6,8 +6,13 @@ hypervisor_defaults:
   local_network:
     name: isolated
     addresses:
-      - 192.168.201.0/24
-    dhcp: true
+      - "192.168.201.0/24"
+      #- "2001:db8:ca2:2::1/64"
```
Comment (rustydb) [inline dev/inventory.yaml:10]:
```suggestion
```

**Review by rustydb — APPROVED** [review]

File: modules/network/variables.tf [hunk 2]
```diff
@@ -37,7 +37,7 @@ variable "autostart" {
 variable "mode" {
   description = "The network mode"
   type        = string
-  default     = "none"
+  default     = "nat"
```
Comment (manderson-hpe) [inline modules/network/variables.tf:40]: Do we need NAT on this? If we have the macvtap interfaces in dev then `none` is fine. this doesn't hurt anything, just might be redundant. Perhaps are we removing the macvtap in local dev and just doing nat for simplification?

**Review by manderson-hpe — APPROVED** [review]
Just one question. the locals are a bit difficult to follow in PR form but everything else looks decent.

File: dev/inventory.yaml (hunk 1 above)
Comment (muresan) [reply dev/inventory.yaml:10]: 2001:db8::/32 - "IPv6 Address Prefix Reserved for Documentation”, RFC 3849

File: modules/network/variables.tf (hunk 2 above)
Comment (muresan) [reply modules/network/variables.tf:40]: I had nat for a case where I did not have macvtap enabled and it sort of lingered. It's still dev.

**Pull request merged**